api/src/paths/administrative/security/submission/{submissionId}/index.test.ts

```diff
@@ -0,0 +1,155 @@
+    };
+    mockReq.body = {
+      applyRuleIds: [4, 5],
+      removeRuleIds: [6, 7]
+    };
+
+    const requestHandler = patchSecurityRulesOnSubmission();
+
+    await requestHandler(mockReq, mockRes, mockNext);
+
+    expect(patchSecurityStub).to.have.been.calledOnceWith(1, [4, 5], [6, 7]);
+    expect(mockRes.statusValue).to.equal(204);
+  });
+
+  it('should call service.patchSecurityRulesOnSubmission with empty remove rule IDs', async () => {
+    const dbConnectionObj = getMockDBConnection({
+      commit: sinon.stub(),
+      rollback: sinon.stub(),
+      release: sinon.stub()
+    });
+
+    sinon.stub(db, 'getDBConnection').returns(dbConnectionObj);
+
+    const patchSecurityStub = sinon.stub(SecurityService.prototype, 'patchSecurityRulesOnSubmission').resolves();
+
+    const { mockReq, mockRes, mockNext } = getRequestHandlerMocks();
+
+    mockReq.params = {
+      submissionId: '2'
+    };
+    mockReq.body = {
+      applyRuleIds: [4, 5],
+      removeRuleIds: []
+    };
+
+    const requestHandler = patchSecurityRulesOnSubmission();
+
+    await requestHandler(mockReq, mockRes, mockNext);
+
+    expect(patchSecurityStub).to.have.been.calledOnceWith(2, [4, 5], []);
+    expect(mockRes.statusValue).to.equal(204);
+  });
+
+  it('should call service.patchSecurityRulesOnSubmission with empty apply rule IDs', async () => {
+    const dbConnectionObj = getMockDBConnection({
+      commit: sinon.stub(),
+      rollback: sinon.stub(),
+      release: sinon.stub()
+    });
+
+    sinon.stub(db, 'getDBConnection').returns(dbConnectionObj);
+
+    const patchSecurityStub = sinon.stub(SecurityService.prototype, 'patchSecurityRulesOnSubmission').resolves();
+
+    const { mockReq, mockRes, mockNext } = getRequestHandlerMocks();
+
+    mockReq.params = {
+      submissionId: '3'
+    };
+    mockReq.body = {
+      applyRuleIds: [],
+      removeRuleIds: [6, 7]
+    };
+
+    const requestHandler = patchSecurityRulesOnSubmission();
+
+    await requestHandler(mockReq, mockRes, mockNext);
+
+    expect(patchSecurityStub).to.have.been.calledOnceWith(3, [], [6, 7]);
+    expect(mockRes.statusValue).to.equal(204);
+  });
+
+  it('should call service.patchSecurityRulesOnSubmission with both empty arrays', async () => {
+    const dbConnectionObj = getMockDBConnection({
+      commit: sinon.stub(),
+      rollback: sinon.stub(),
+      release: sinon.stub()
+    });
+
+    sinon.stub(db, 'getDBConnection').returns(dbConnectionObj);
+
+    const patchSecurityStub = sinon.stub(SecurityService.prototype, 'patchSecurityRulesOnSubmission').resolves();
+
+    const { mockReq, mockRes, mockNext } = getRequestHandlerMocks();
+
+    mockReq.params = {
+      submissionId: '4'
+    };
+    mockReq.body = {
+      applyRuleIds: [],
+      removeRuleIds: []
+    };
+
+    const requestHandler = patchSecurityRulesOnSubmission();
+
+    await requestHandler(mockReq, mockRes, mockNext);
+
+    expect(patchSecurityStub).to.have.been.calledOnceWith(4, []);
```

Comment:
[NIT] The `patchSecurityRulesOnSubmission` service method takes 3 arguments (`submissionId`, `applyRuleIds`, `removeRuleIds`).

Consider using `calledOnceWithExactly` to verify the full method signature:

```suggestion
    expect(patchSecurityStub).to.have.been.calledOnceWithExactly(4, [], []);
```